Feature/243/unsuccesful login

vvichuuma opened the pull request:
### Ticket Number:243



### Describe The Problem Being Solved: Fixed the Login button , when a user clicks on the login button , the validation pops up multiple times. I fixed it so that it appears only once. 



### Checklist For Submitter

* [ x] I have documented all new functionality
* [ x
![Login](https://user-images.githubusercontent.com/40339176/57664950-ab05d400-75bf-11e9-9cb2-b67e4757afba.png)
] I have screenshotted new UI changes and attached them to this PR

H12 commented on frontend/src/components/SignIn/SignIn.js:
```diff
@@ -31,23 +34,32 @@ class SignIn extends Component {
+          }, this.setState({
+            err: 'Please make sure you have the correct email and password!',
+          }));
+          // const parentForm = document.getElementsByTagName('form')[1];
```
Please remove commented-out code that is no longer needed.

H12 commented on frontend/src/components/SignIn/SignIn.js:
```diff
@@ -12,6 +12,9 @@ class SignIn extends Component {
       isPasswordVisible: false,
       email: '',
       password: '',
+      errorText: 'Please Vish is a programmer!',
```
It looks like `errorText` isn't being used anywhere, so I think we can delete this line.

H12 commented on frontend/src/components/SignIn/SignIn.js:
```diff
@@ -31,23 +34,32 @@ class SignIn extends Component {
-          errorDiv.innerHTML = 'Unable to log in. Check your email and password and try again.';
-          parentForm.insertAdjacentElement('beforeend', errorDiv);
+          this.setState({
+            flags: true,
```
It doesn't look like `flags` is being used anywhere. Can you make this work by only using `err`?

H12 commented on frontend/src/components/SignIn/SignIn.js:
```diff
@@ -12,6 +12,9 @@ class SignIn extends Component {
       password: '',
+      errorText: 'Please Vish is a programmer!',
+      flags: false,
+      err: '',
```
NitPick: Let's come up with a more descriptive name for `err`.

vhoof commented on frontend/src/components/SignIn/SignIn.js:
```diff
@@ -91,6 +103,7 @@ class SignIn extends Component {
           </div>
 
         </form>
+        <p>{err}</p>
```
Should this `p` element have the `errorMessage` class to ensure it gets styled correctly?

vhoof commented:
You'll probably want to `git pull origin staging` or rebase before you do more work, too.

vvichuuma commented:
Added modifications to the Error message. Changed the color of the message.  Got rid of the extra comments and flag statements.
![error_1](https://user-images.githubusercontent.com/40339176/58215484-b479f500-7cbf-11e9-9dd7-4eb3704137cf.png)
![error_2](https://user-images.githubusercontent.com/40339176/58215490-bb086c80-7cbf-11e9-9cee-efb947124582.png)

SamDesai333 commented on frontend/src/components/SignIn/SignIn.css:
```diff
@@ -37,3 +37,7 @@
 }
+
+.errorValidation {
+  color: red;
```
Good idea changing the color of the text! Makes the error seem more apparent.

vhoof replied:
This looks good, but maybe we want to delete the `.errorMessage` CSS since we're not using that anymore?

SamDesai333 commented on frontend/src/components/SignIn/SignIn.js:
```diff
@@ -12,6 +12,7 @@ class SignIn extends Component {
       isPasswordVisible: false,
       email: '',
       password: '',
+      error: '',
```
Good job adding in an empty string for error and updating the state!

vhoof replied:
Can we change the name of this variable to something like `validationError` or `validationErrorText`? That way it'll be a little easier to understand/distinguish from other things called `error`?

vvichuuma replied:
Yeah Bryan..will worked on it.

vvichuuma commented:
![validationError](https://user-images.githubusercontent.com/40339176/58595115-370d3200-8235-11e9-8866-f8a337f5aa66.png)
Changed the error text to validationErrortext. removed the extra css style.

vhoof commented on frontend/src/components/RecipeSearchList/RecipeSearchList.css:
```diff
@@ -0,0 +1,3 @@
+#cards-wrapper{
+  border: 1px solid red;
```
What's this change for?

vhoof commented on frontend/src/components/SignIn/SignIn.js:
```diff
@@ -34,19 +35,19 @@ class SignIn extends Component {
+
+            validationErrorText: 'Please make sure you have the correct email and passwords!',
+
+          });
```
I think the two blank lines are probably unnecessary, and makes reading the function call a little weird.

vhoof commented on frontend/src/components/SignIn/SignIn.js:
```diff
@@ -34,19 +35,19 @@ class SignIn extends Component {
   render() {
     const { changeAuthorizedState } = this.props;
-    const { isPasswordVisible } = this.state;
+
```
I'd also remove this blank line.
